DependencyInjection/HttplugExtension.php

```diff
@@ -87,6 +87,10 @@ public function load(array $configs, ContainerBuilder $container)
         $this->configureClients($container, $config);
         $this->configurePlugins($container, $config['plugins']); // must be after clients, as clients.X.plugins might use plugins as templates that will be removed
         $this->configureAutoDiscoveryClients($container, $config);
+
+        if (!$config['register_default_client']) {
+            $container->removeDefinition('httplug.client.default');
```

Comment:
your goal is to prevent autowiring from happening, right? would it not be better to have an `autowire_client` flag and remove the alias with the client class name instead of the `default` service? with this, you would get an error that the alias is invalid, no?

Reply:
I think the difference here is Symfony 3.4 and 4+. 

With Symfony 3.4: when you have a `default` service, it will just auto wire when you use its class name. 
I believe this changed in 4+ where you have to explicitly set the name of the service to the class name.

I faced this issue in a SF 3.4 application.